scripts/react-native-xcode.sh

```diff
@@ -194,14 +194,12 @@ else
   if [[ $EMIT_SOURCEMAP == true ]]; then
     EXTRA_COMPILER_ARGS="$EXTRA_COMPILER_ARGS -output-source-map"
   fi
-  HBC_FILE="$CONFIGURATION_BUILD_DIR/$(basename $BUNDLE_FILE)"
-  "$HERMES_CLI" -emit-binary $EXTRA_COMPILER_ARGS -out "$HBC_FILE" "$BUNDLE_FILE"
-  mv "$HBC_FILE" "$DEST/"
-  BUNDLE_FILE="$DEST/main.jsbundle"
+  "$HERMES_CLI" -emit-binary $EXTRA_COMPILER_ARGS -out "$DEST/main.jsbundle" "$BUNDLE_FILE" 
   if [[ $EMIT_SOURCEMAP == true ]]; then
-    HBC_SOURCEMAP_FILE="$HBC_FILE.map"
+    HBC_SOURCEMAP_FILE="$BUNDLE_FILE.map"
     "$NODE_BINARY" "$COMPOSE_SOURCEMAP_PATH" "$PACKAGER_SOURCEMAP_FILE" "$HBC_SOURCEMAP_FILE" -o "$SOURCEMAP_FILE"
   fi
+  BUNDLE_FILE="$DEST/main.jsbundle"
 fi
 
 if [[ $DEV != true && ! -f "$BUNDLE_FILE" ]]; then
```

Comment:
I don't think we need this condition BTW. It's invalid.

First of all, if the `main.jsbundle` is not generated in `$DEV = true` but with `$FORCE_BUNDLING = true`, this will not report any issues.

Also, at this point, we always want to check whether bundle exists. If we are not bundling (and there's nothing to check), we're going to exit early here: https://github.com/facebook/react-native/pull/29914/commits/cb01604e09403e0738311060a26375333dca5b39#diff-f1a28d5a04c65bf04339b2fd16a6f56fR35-R57

I believe this code is left-over from previous form of this file and the `if` statement should be removed

Reply:
I could most likely send a follow-up PR to address that, but I am not sure whether we want to do it for just a single line.